examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,137 @@
+import typer
+from rich.prompt import Prompt
+import langroid as lr
+from agents import (
+    create_pro_ai_agent, create_con_ai_agent,
+    create_pro_ip_agent, create_con_ip_agent,
+    create_pro_bias_agent, create_con_bias_agent,
+    create_pro_edu_agent, create_con_edu_agent,
+    create_feedback_agent,
+)
+from tasks import (
+    create_pro_ai_task, create_con_ai_task,
+    create_pro_ip_task, create_con_ip_task,
+    create_pro_bias_task, create_con_bias_task,
+    create_pro_edu_task, create_con_edu_task,
+    create_feedback_task,
+)
+from config import get_global_settings, get_base_llm_config, is_llm_delegate
+import logging
+
+# Set up the Typer application
+app = typer.Typer()
+
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+lr.utils.logging.setup_colored_logging()
+
+# Mappings for dynamic task and agent creation
+DEBATE_TOPICS = {
+    "1": ("AI in Healthcare", "pro_ai", "con_ai"),
+    "2": ("AI and Intellectual Property", "pro_ip", "con_ip"),
+    "3": ("AI and Societal Biases", "pro_bias", "con_bias"),
+    "4": ("AI as an Educator", "pro_edu", "con_edu"),
+}
+
+
+def setup_agents_and_tasks(debug=False, nocache=False):
+    """Sets up agents and tasks dynamically."""
+    logger.info("Setting up agents and tasks...")
+    global_settings = get_global_settings(debug, nocache)
+    lr.utils.configuration.set_global(global_settings)
+    base_llm_config = get_base_llm_config()
+
+    agents = {}
+    tasks = {}
+
+    for _, (topic_name, pro_key, con_key) in DEBATE_TOPICS.items():
+        agents[pro_key] = globals()[f"create_{pro_key}_agent"](base_llm_config)
```

Comment:
no need for globals() etc.
If you have system msgs keyed by topic type, you can directly call the generic `create_agent` with the appropriate system msg here, rather than calling the topic-specific agent-creator. Imagine if there are 100 different topics, you wouldn't want to have 100 different specialized agent-definition fns -- instead, you'd have 100 sys msgs keyed by topic, and call the generic agent-defn fn, with the specific sys msg

Reply:
this was a good suggestion. completely refactored to dynamically generate agents based on system_messages.json.